virttest/qemu_devices/qdevices.py

```diff
@@ -756,11 +756,13 @@ def _convert_blkdev_args(args):
         :rtype: dict
         """
         new_args = dict()
+        keep_original_type = ("detect-zeroes", )
         for key, value in six.iteritems(args):
-            if value in ('on', 'yes'):
-                value = True
-            elif value in ('off', 'no'):
-                value = False
+            if key not in keep_original_type:
+                if value in ('on', 'yes'):
+                    value = True
+                elif value in ('off', 'no'):
+                    value = False
```

Comment:
```suggestion
            if key in ("detect-zeroes", ):
                  continue
            if value in ('on', 'yes'):
                value = True
            elif value in ('off', 'no'):
                value = False
```
Just add new codes and don't modify the previous codes.

Reply:
Hi @nickzhq 
There is a regression issue with my suggestion code, it will miss(drop) the option `detect-zeroes` from the blockdev options.  I submitted a patch to fix it: https://github.com/avocado-framework/avocado-vt/pull/3771
Sorry for the inconvenience @nickzhq  @zhencliu @qingwangrh